build.yml

```diff
@@ -153,6 +153,14 @@ jobs:
         targetFolder: '$(Build.StagingDirectory)/src/_'
 
     - ${{ parameters.pre_build }}
+    
+    - script: |
+        source /opt/ros/$CI_ROSDISTRO/setup.sh

```

Comment:
Do you want to probe first to see if setup.sh is there? We can throw a more readable error if it isn't.

Reply:
I am trying to stage a failure like this and the pipeline shows:

```
line 1: /opt/ros/melodic/setup.sh: No such file or directory
```
I think it is readable enough and we can save a probe here.